SEED-platform/seed
# Major dependency update

#### What's this PR do?
- **Updates nearly all Python dependencies (except Django, and ones that rely on newer versions of Django/Python)**
- Adds a `manage.py flush_db` command that wipes all database contents (and restarts table sequences) except for static tables (EEEJ and Uniformat)
- Removes vestigial `django-autoslug` dependency
- Removes unused code/files
- Removes deprecated `NullBooleanField`
- Fixes sphinx spelling issues
- Fixes sphinx-build on Windows
- Updates tests
- Fixed the following errors when visiting the Swagger page
    - `<class 'seed.views.v3.sensors.SensorViewSet'> is not compatible with schema generation`
    - `<class 'seed.views.v3.tax_lot_properties.TaxLotPropertyViewSet'> is not compatible with schema generation`
- Fixes new Ruff issues
- Fixes typos
- Improves `npm test` to support both port 80 and 8000

#### How should this be manually tested?
Monkey test, tests should pass
For local testing, you'll (obviously) need to run `pip install -r requirements/test.txt`
